src/server/scripts/EasternKingdoms/Deadmines/boss_vancleef.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2008-2017 TrinityCore <http://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "deadmines.h"
+
+enum VanCleefData
+{
+    NPC_BLACKGUARD = 636,
+
+    SPELL_DUAL_WIELD = 674,
+    SPELL_THRASH = 12787,
+    SPELL_VANCLEEFS_ALLIES = 5200,
+
+    PHASE_NORMAL = 1,
+    PHASE_66_PCT = 2,
+    PHASE_50_PCT = 3,
+    PHASE_33_PCT = 4,
+    PHASE_25_PCT = 5
+};
+
+enum Speech
+{
+    SAY_AGGRO  = 0,    // None may challenge the Brotherhood!
+    SAY_ONE    = 1,    // Lapdogs, all of you!
+    SAY_SUMMON = 2,    // calls more of his allies out of the shadows.
+    SAY_TWO    = 3,    // Fools! Our cause is righteous!
+    SAY_KILL   = 4,    // And stay down!
+    SAY_THREE  = 5     // The Brotherhood shall prevail!
+};
+
+// From TDB 335.63
+Position const BlackguardPositions[] =
+{
+        { -78.2791f, -824.784f, 40.0007f, 2.93215f },
+        { -77.8071f, -815.097f, 40.0188f, 3.26377f }
+};
+
+struct boss_vancleef : public BossAI
+{
+    public:
+        boss_vancleef(Creature* creature) : BossAI(creature, 0) { }
```

Comment:
magic number
should have a DATA_BOSS_VANCLEEF or something like that in deadmines.h
Btw deadmines.h should be called Deadmines.h

Reply:
Yeah I wondered about that but since there's only one boss with BossAI I didn't know if it would be necessary.

Reply:
yes, is

Reply:
That's fixed now. I added Mr Smite since he's currently the only other boss scripted, but doesn't use BossAI (that probably needs to be rewritten too)